src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,276 @@
+import { AfterContentChecked, Directive, ElementRef, Input, OnChanges, OnDestroy, Renderer2, SimpleChanges } from '@angular/core';
+import { ColorService } from './../../services/color/color.service';
+import { ThemeColor } from './../../services/color/theme-color';
+import { ContrastService } from './../accessibility/contrast-ratio/contrast.service';
+
+export type BadgeVerticalPosition = 'above' | 'below' | 'top' | 'bottom';
+
+export type BadgeHorizontalPosition = 'before' | 'after' | 'left' | 'right';
```

Comment:
@donna-marijne Do you think we need both 'above/below' and 'top/bottom' or can we just have one set as they both mean the same thing?

As for `before` and `after` and `left` and `right` we might want only one of these, however there is an additional future difference here that we should consider if we do decide to have both. Left and right are explicit directional values, however before and after are not, they are dependent on the language directionality, eg. in ltr language before is left, but in rtl language before is right. Just a thought.

Reply:
No we don't need duplicates, and Ashley's point about RTL languages is well made. Although I'm not aware of anyone using UX Aspects in an RTL locale, it's good forward planning to establish APIs which won't cause problems if/when we do have to address it.

It's good to be accommodating of things like synonyms when processing input that's coming from a user, but this is a developer API with documentation and TypeScript type definition. Extra checks for developer convenience unfortunately turn into performance problems when you have a page with 10,000 input bindings.

Reply:
Thanks again for feedback, so very good points made especially about direction. I've removed the duplicates.